core/sdk-core/src/main/java/software/amazon/awssdk/core/client/handler/ClientExecutionParams.java

```diff
@@ -167,6 +168,18 @@ public ClientExecutionParams<InputT, OutputT> withOperationName(String operation
         return this;
     }
 
+    public String getServiceProtocol() {
+        return serviceProtocol;
```

Comment:
Should this be an enum?

Reply:
Hmm the division between sdk and aws is a little bit dodgy. Service protocol feels more aws but it probably needs to go in sdk core? Maybe we can wait till we have stricter boundaries between them unless you have a good suggestion of placement.

Reply:
Discussed offline, adding `SdkProtocolMetadata` and `AwsProtocolMetadata` where we can put additional protocol information in future